### ISSUE TITLE
Delete cluster fails with a private topology due to 'AuthFailure' when deleting elastic ip resource

### ISSUE THREAD

**Issue by lewistuff** [kubernetes/kops#938]
I created a new cluster with `--topology private`, as below

> kops create cluster \
>   --state $KOPS_STATE_STORE \
>   --node-count 2 \
>   --zones eu-west-1a,eu-west-1b \
>   --master-zones eu-west-1a \
>   --cloud aws \
>   --node-size t2.medium \
>   --master-size m3.large \
>   --topology private \
>   --networking weave \
>   --ssh-public-key=$PUB_KEY \
>   -v 10 \
>   $CLUSTER_NAME \
>   --yes
> 

**Problem**
When trying to delete with `kops delete cluster $CLUSTER_NAME --yes` I hit the following issue.
> I1118 13:37:16.813541   17638 delete_cluster.go:769] unexpected aws error code: "AuthFailure"
> elastic-ip:eipalloc-12af8a76    error deleting resource, will retry: error deleting elastic ip "52.51.151.34": AuthFailure: You do not have permission to access the specified resource.

**Workaround**
To resolve, I had to manually delete the associated NAT gateways, and then I was able to successfully re-run `kops` to complete the clean-up. 

**Note**, 
1. I could not disassociate/release the EIPs through the AWS console,
2. The "AuthFailure" is a red herring, as I had sufficient permissions.

**Comment by chrislovecnm** [comment]
Did `kops` timeout on the delete?  I have seen where the aws api can be a bit pissy.

**Comment by kris-nova** [comment]
I'd be willing to say that is what happened - I bumped the timeout once - but we can turn it up even higher.... 99% of the time waiting a few minutes and re running the command will correct this issue

**Comment by kris-nova** [comment]
I'll get a PR out that allows the user to define a custom timeout - so we can start seeing how long this needs to be

**Comment by lewistuff** [comment]
On the delete on the NAT gateway, or the Elastic IP? I only observed errors for the latter.

I left it running through retries, and it timed out. I tried a second time some time later and it failed again with the timeout - after a large number of retries. 

Trying to delete the elastic IP resource from the AWS console also failed until I manually deleted the NAT gateway. I confirmed with AWS support this was the reason I was receiving the "AuthFailure".
